src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultBoundAttributeDescriptorBuilder.cs

```diff
@@ -219,10 +219,10 @@ private void Validate(ref PooledHashSet<RazorDiagnostic> diagnostics)
                 diagnostics.Add(diagnostic);
             }
 
-            StringSegment name = Name;
-            if (isDirectiveAttribute && name.StartsWith("@", StringComparison.Ordinal))
+            var name = Name.AsSpan();
+            if (isDirectiveAttribute && name[0] == '@')
```

Comment:
For anyone else reviewing this code the `IsNullOrWhitespace` on line 199 also checks for empty string cases. I had a whole comment written up here about length 0 cause the name doesn't sae `IsNullEmptyOrWhitespace` but alas it does check for empty.